msal/src/main/java/com/microsoft/identity/client/internal/controllers/BrokerContentProviderStrategy.java

```diff
@@ -0,0 +1,446 @@
+                            .isSuccessful(true)
+            );
+            Logger.info(TAG + methodName, "Received successful result from broker");
+            return result;
+        } else {
+            final String message = "Failed to get result from Broker Content Provider, cursor is null";
+            Logger.error(TAG + methodName, message, null);
+            Telemetry.emit(
+                    new BrokerEndEvent()
+                            .putAction(methodName)
+                            .isSuccessful(false)
+                            .putErrorDescription(message)
+            );
+            throw new BrokerCommunicationException(message, null);
+        }
+    }
+
+    @Override
+    String hello(@NonNull CommandParameters parameters) throws BaseException {
+        final String methodName = "helloWithContentProvider";
+
+        return performContentProviderOperation(parameters, new ContentProviderOperation<CommandParameters, String>() {
+            @Override
+            public Bundle getRequestBundle(CommandParameters parameters) {
+                return mRequestAdapter.getRequestBundleForHello(parameters);
+            }
+
+            @Override
+            public String getMethodName() {
+                return methodName;
+            }
+
+            @Override
+            public String getUriPath() {
+                return HELLO_PATH;
+            }
+
+            @Override
+            public String getResultFromBundle(Bundle resultBundle) throws BaseException {
+                return mResultAdapter.verifyHelloFromResultBundle(resultBundle);
+            }
+        });
+    }
+
+    @Override
+    Intent getBrokerAuthorizationIntent(@NonNull final InteractiveTokenCommandParameters parameters,
+                                        @Nullable final String negotiatedBrokerProtocolVersion)
+            throws BaseException {
+        final String methodName = "getBrokerAuthorizationIntentForContentProvider";
+
+        return performContentProviderOperation(
+                parameters,
+                new ContentProviderOperation<InteractiveTokenCommandParameters, Intent>() {
+
+                    @Override
+                    public Bundle getRequestBundle(InteractiveTokenCommandParameters parameters) {
+                        return null; // broker returns us an intent based on calling uid , no request string needed
+                    }
+
+                    @Override
+                    public String getMethodName() {
+                        return methodName;
+                    }
+
+                    @Override
+                    public String getUriPath() {
+                        return ACQUIRE_TOKEN_INTERACTIVE_PATH;
+                    }
+
+                    @Override
+                    public Intent getResultFromBundle(Bundle resultBundle) throws BaseException {
+                        return mRequestAdapter.getRequestIntentForAcquireTokenInteractive(
```

Comment:
Is this doing the same thing as completeInteractiveRequestIntent()?

(Other strategies are using that, let's refactor in such a way that they're doing the same thing - either using that function or keep using this).

Reply:
Nope, it's not exactly doing the same thing. We can get an `Intent` as a result using `ContentProvider` . I mean technically we don't need an `Intent` here for other strategies as well, all we need is a `bundle` or `data` to launch the `AccountChooserActivity`. So I consolidated constructing the `bundle` logic in `broker` , but it returns an `intent` for `auth` and `accountmanager` strategies and `bundle` here , so we need to construct the `intent` and launch the activity